src/scalar/dusk.rs

```diff
@@ -40,52 +39,16 @@ impl Serializable<32> for Scalar {
     /// Converts an element of `Scalar` into a byte representation in
     /// little-endian byte order.
     fn to_bytes(&self) -> [u8; Self::SIZE] {
-        // Turn into canonical form by computing
-        // (a.R) / R = a
-        let tmp = Scalar::montgomery_reduce(self.0[0], self.0[1], self.0[2], self.0[3], 0, 0, 0, 0);
-
-        let mut res = [0; Self::SIZE];
-        res[0..8].copy_from_slice(&tmp.0[0].to_le_bytes());
-        res[8..16].copy_from_slice(&tmp.0[1].to_le_bytes());
-        res[16..24].copy_from_slice(&tmp.0[2].to_le_bytes());
-        res[24..32].copy_from_slice(&tmp.0[3].to_le_bytes());
-
-        res
+        self.to_bytes()
     }
 
     /// Attempts to convert a little-endian byte representation of
     /// a scalar into a `Scalar`, failing if the input is not canonical.
     fn from_bytes(buf: &[u8; Self::SIZE]) -> Result<Self, Self::Error> {
-        let mut s = [0u64; 4];
-
-        s.iter_mut()
-            .zip(buf.chunks_exact(8))
-            .try_for_each(|(s, b)| {
-                <[u8; 8]>::try_from(b)
-                    .map(|b| *s = u64::from_le_bytes(b))
-                    .map_err(|_| BytesError::InvalidData)
-            })?;
-
-        // Try to subtract the modulus
-        let (_, borrow) = sbb(s[0], MODULUS.0[0], 0);
-        let (_, borrow) = sbb(s[1], MODULUS.0[1], borrow);
-        let (_, borrow) = sbb(s[2], MODULUS.0[2], borrow);
-        let (_, borrow) = sbb(s[3], MODULUS.0[3], borrow);
-
-        // If the element is smaller than MODULUS then the
-        // subtraction will underflow, producing a borrow value
-        // of 0xffff...ffff. Otherwise, it'll be zero.
-        if (borrow as u8) & 1 != 1 {
-            return Err(BytesError::InvalidData);
+        match Self::from_bytes(buf).into_option() {
+            Some(s) => Ok(s),
+            None => Err(BytesError::InvalidData),
```

Comment:
I think you can use `ok_or` here:
```suggestion
Self::from_bytes(buf).into_option().ok_or(Err(BytesError::InvalidData))
```